src/Compression/CompressedReadBufferBase.cpp

```diff
@@ -337,7 +338,9 @@ void CompressedReadBufferBase::addDiagnostics(Exception & e) const
     if (auto * seekable_in = dynamic_cast<SeekableReadBuffer *>(compressed_in))
         current_pos = seekable_in->tryGetPosition();
     UInt8 header_size = ICompressionCodec::getHeaderSize();
-    String header_hex = hexString(own_compressed_buffer.data(), std::min(own_compressed_buffer.size(), sizeof(Checksum) + header_size));
+    String header_hex = own_compressed_buffer_header_init ?
+        hexString(own_compressed_buffer.data(), std::min(own_compressed_buffer.size(), sizeof(Checksum) + header_size)) :
+        String("<uninitialized>");
```

Comment:
Perhaps it's better to write a comment here saying that we shouldn't print uninitialized memory.